pkg/ctl/cmdutils/filter/iamserviceaccount_filter.go

```diff
@@ -57,6 +57,9 @@ func (f *IAMServiceAccountFilter) SetExcludeExistingFilter(stackManager serviceA
 
 	if !overrideExistingServiceAccounts {
 		err := f.ForEach(serviceAccounts, func(_ int, sa *api.ClusterIAMServiceAccount) error {
+			if api.IsEnabled(sa.RoleOnly) {
```

Comment:
This means with `roleOnly`, we'll always create the role, right? Can we also add that to the docs?

Reply:
The use case is that the service account already exists in the cluster or is created later by some other tool. But by default eksctl will not create iam roles for service accounts if the service account already exists. But with RoleOnly, the service account will never be created and overwritten by eksctl, so we should also not filter for them. I think this is the behavior a user expects when using the roleOnly option. Otherwise a user has to specify `--override-existing-serviceaccounts` and `--role-only` at the same time which is confusing, because the service accounts are not overwritten.